src/QsFmt/SPECIFICATION.md

```diff
@@ -0,0 +1,68 @@
+﻿# Q# Formatter Design Specification Document
+
+QsFmt is a source code formatter for Q#.
+
+## Where to get the formatter
+
+The `Microsoft.Quantum.QSharp.Formatter` NuGet package will be distributed with the rest of Q# on NuGet.org.
+As a dotnet tool, it is installed with the command `dotnet tool install Microsoft.Quantum.QSharp.Formatter`.
+
+## How to use the command line formatter
+
+After installing the tool the command `qsfmt` is used to run the formatter.
+There are two commands supported for the tool: 'format' and 'update'.
+The 'format' command runs the tool as a formatter for Q# code, running only those transformation rules that update whitespace and affect indentation. The underlying code will not be changed.
+The 'update' command allows the tool to be used as a means of updating old Q# code, replacing deprecated syntax with newer supported syntax. These transformation rules change the actual code given to them, and may occasionally fail.
+These two commands work separately and can't be run together. If the user wishes to both format and update their code, they would need to run the tool twice, first with 'update' and then with 'format'.
+Both of these commands should be idempotent, meaning that iterated uses of the commands will not continue to have effects on the inputs.
+
+The tool will initially only support the 'update' command, as the formatting functionality is still being worked on. The 'format' command will be unavailable until it is supported.
+
+## Input and Output
+Input files to the formatter will be the last argument(s) to the tool.
+At least one input argument is expected. If no inputs are given, usage information is printed.
+Inputs can be given as one to many paths to directories, files, or any combination. In directories specified, all files with the '.qs' extension found will be taken as input.
+Basic wild cards may be used in the paths.
+The `--recurse` option can be specified to process the input directories recursively for input files. You can specify this for all input directories, not per-directory.
+The `--backup` option can be specified to create backup files for all input files with the original content in them.
```

Comment:
I think backup files should be created by default so the user can rollback the changes if needed

Reply:
So instead of a `--backup` option, we would have a `--no-backup` option to instruct the tool to not create backup files it would otherwise normally create? That seems reasonable to me. Let's see what others think on this.

Reply:
Creating a backup file for every formatted file is a fairly intrusive operation since it effectively doubles the size of every source folder, which can make navigating harder. With version control, backup files are also mostly redundant. Also, my experience with other formatters is that they don't create backup files by default.

Reply:
Moving this conversation to the spec document comments section. Please pick up this conversation there.